File: 429-n-ary-tree-level-order-traversal/429-n-ary-tree-level-order-traversal.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def levelOrder(self, root: 'Node') -> List[List[int]]:
        res,q = [],[]
        if not root: return res
        q.append(root)
        while q:
            n = len(q)
            level = []
            while(n>0):
                curr = q[0]
                q.pop(0)
                level.append(curr.val)
                n -= 1
                for i in curr.children:
                    q.append(i)
            res.append(list(level))
        return res

File: 429-n-ary-tree-level-order-traversal/429-n-ary-tree-level-order-traversal.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;
    Node() {}
    Node(int _val) {
        val = _val;
    }
    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    vector<vector<int>> levelOrder(Node* root) {
        vector<vector<int>> res;
        queue<Node*> q;
        if(!root) return res;
        q.push(root);
        while(!q.empty()){
            int n =q.size();
            vector<int> level;
            while(n>0){
                Node* curr = q.front();
                q.pop();
                level.push_back(curr->val);
                n--;
                for(auto child: curr->children)
                    q.push(child);
            }
            res.push_back(level);
        }
        return res;
    }
};